### PR TITLE
feat: upgrade to distribution-spec v1.1.0

### LINKED ISSUE #709: Upgrade to `distribution-spec v1.1.0`
[`distribution-spec v1.1.0`](https://github.com/opencontainers/distribution-spec/releases/tag/v1.1.0) has just been released.

No specific changes related to `oras-go` but to
- Mark all RC related APIs `deprecated`.
  - Introduce `PackManifestVersion1_1` and deprecate [`PackManifestVersion1_1_RC4`](https://pkg.go.dev/oras.land/oras-go/v2#PackManifestVersion)
  - Update golang examples to use `PackManifestVersion1_1`
- Remove `-rc` in URLs referenced in the comments

### PR DESCRIPTION
**Pull request by wangxiaoxuan273**
Resolves #709

### CODE REVIEW

File: pack.go [hunk 1]
```diff
@@ -71,7 +71,12 @@ const (
 	// PackManifestVersion1_1_RC4 represents the OCI Image Manifest defined
 	// in image-spec v1.1.0-rc4.
 	// Reference: https://github.com/opencontainers/image-spec/blob/v1.1.0-rc4/manifest.md
-	PackManifestVersion1_1_RC4 PackManifestVersion = 2
+	PackManifestVersion1_1_RC4 PackManifestVersion = PackManifestVersion1_1
```
Comment (shizhMSFT) [inline pack.go:74]: This constant should be marked as deprecated.

File: pack.go [hunk 2]
```diff
@@ -122,7 +127,7 @@ func PackManifest(ctx context.Context, pusher content.Pusher, packManifestVersio
 	switch packManifestVersion {
 	case PackManifestVersion1_0:
 		return packManifestV1_0(ctx, pusher, artifactType, opts)
-	case PackManifestVersion1_1_RC4:
+	case PackManifestVersion1_1:
 		return packManifestV1_1_RC4(ctx, pusher, artifactType, opts)
```
Comment (shizhMSFT) [inline pack.go:131]: The method name `packManifestV1_1_RC4` should be renamed to `packManifestV1_1`.

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

File: pack.go (hunk 1 above)
Comment (wangxiaoxuan273) [reply pack.go:77]: deprecated this constant.

File: pack.go (hunk 2 above)
Comment (wangxiaoxuan273) [reply pack.go]: renamed the method.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**